azurerm/internal/services/subscription/subscription_resource.go

```diff
@@ -0,0 +1,499 @@
+package subscription
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-11-01/subscriptions"
+	subscriptionAlias "github.com/Azure/azure-sdk-for-go/services/subscription/mgmt/2020-09-01/subscription"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	azValidate "github.com/terraform-providers/terraform-provider-azuread/azuread/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/subscription/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/subscription/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+var (
+	billingScopeEnrollmentFmt = "/providers/Microsoft.Billing/billingAccounts/%s/enrollmentAccounts/%s"
+	billingScopeMCAFmt        = "/providers/Microsoft.Billing/billingAccounts/%s/billingProfiles/%s/invoiceSections/%s"
```

Comment:
as discussed offline - we should make data sources for these (and make the fields `{thing}_id`), even if initially these are just templating the ID's whilst the SDK is fixed - since this means if other formats are added later we can more easily validate and differentiate these, these presumably should work:

https://docs.microsoft.com/en-us/rest/api/billing/2019-10-01-preview/invoicesections/get
https://docs.microsoft.com/en-us/rest/api/billing/2019-10-01-preview/enrollmentaccounts